src/code_805A120.c

```diff
@@ -0,0 +1,1474 @@
+    local_20 = local_20 == 0;
+    return local_20;
+}
+
+bool8 FlyMoveAction(struct Entity * pokemon, struct Entity * target, struct Move * move, u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (MoveMatchesChargingStatus(pokemon,move)) {
+      flag = sub_8055640(pokemon,target,move,gUnknown_80F4F5C,param_4) != 0 ? TRUE : FALSE;
+      sub_8079764(pokemon);
+  }
+  else {
+      SetChargeStatusTarget(pokemon,pokemon,STATUS_FLYING,move,*gUnknown_80FACFC);
+      flag = TRUE;
+  }
+  return flag;
+}
+
+bool8 sub_80597F0(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
+{
+    sub_807DF38(pokemon,target,&target->pos,2,GetMoveType(move),sub_8057600(move,param_4));
+    return TRUE;
+}
+
+bool8 DiveMoveAction(struct Entity * pokemon, struct Entity * target, struct Move * move, u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (IsTileGround(GetTileAtEntitySafe(pokemon))) {
+    sub_80522F4(pokemon,target,*gUnknown_80FD128);
+  }
+  else if (MoveMatchesChargingStatus(pokemon,move)) {
+      flag = sub_8055640(pokemon,target,move,gUnknown_80F4F64,param_4) != 0 ? TRUE : FALSE;
+      sub_8079764(pokemon);
+  }
+  else {
+      SetChargeStatusTarget(pokemon,pokemon,STATUS_DIVING,move,*gUnknown_80FAD2C);
+      flag = TRUE;
+  }
+  return flag;
+}
+
+bool8 sub_80598CC(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+    flag = TRUE;
+    if(sub_805727C(pokemon, target, gUnknown_80F4E00))
+    {
+        LowerAccuracyStageTarget(pokemon, target, gUnknown_8106A4C, FALSE);
+    }
+  }
+  return flag;
+}
+
+bool8 StockpileMoveAction(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+    HandleStockpile(pokemon, target);
+    return TRUE;
+}
+
+bool8 sub_8059928(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
+{
+  bool8 flag;
+  s32 iVar2;
+  
+  iVar2 = 1;
+  flag = FALSE;
+  if ((u8)(target->info->chargingStatus - 7) <= 1){
+      iVar2 = 2;
+  }
+  if (sub_8055640(pokemon,target,move,iVar2 << 8,param_4) != 0)
+  {
+    flag = TRUE;
+    if(sub_805727C(pokemon,target,gUnknown_80F4DEC) != 0) {
+        CringeStatusTarget(pokemon,target,FALSE);
+    } 
+  }
+  return flag;
+}
+
+bool8 sub_8059988(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+    flag = TRUE;
+    gUnknown_202F218 = 1;
+  }
+  if (((gUnknown_202F218 != 0) && (gUnknown_202F214 == 2)) &&
+     (sub_805727C(pokemon,target,gUnknown_80F4DC8) != 0)) {
+    PoisonedStatusTarget(pokemon,target,FALSE);
+  }
+  return flag;
+}
+
+bool8 sub_80599EC(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
```

Comment:
This is for regular healing moves like Recover.

Reply:
Same comment above applies here.

Reply:
Going to name to `RecoverMoveAction` for now but we can make it more general later if others do end up calling it